File: src/main/java/org/radarcns/management/service/dto/MinimalSourceDetailsDTO.java
package org.radarcns.management.service.dto;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

/**
 * Created by nivethika on 13-6-17.
 */
public class MinimalSourceDetailsDTO {

    private Long id;
    private Long sourceTypeId;
    private String sourceTypeProducer;
    private String sourceTypeModel;
    private String sourceTypeCatalogVersion;
    private String expectedSourceName;
    private UUID sourceId;
    private String sourceName;
    private Boolean assigned;
    private Map<String, String> attributes = new HashMap<>();

    public Long getId() {
        return id;
    }

    public MinimalSourceDetailsDTO id(Long id) {
        this.id = id;
        return this;
    }

    public Long getSourceTypeId() {
        return sourceTypeId;
    }

    public MinimalSourceDetailsDTO sourceTypeId(Long sourceTypeId) {
        this.sourceTypeId = sourceTypeId;
        return this;
    }

    public String getExpectedSourceName() {
        return expectedSourceName;
    }

    public MinimalSourceDetailsDTO setExpectedSourceName(String expectedSourceName) {
        this.expectedSourceName = expectedSourceName;
        return this;
    }

    public UUID getSourceId() {
        return sourceId;
    }

    public MinimalSourceDetailsDTO sourceId(UUID sourceId) {
        this.sourceId = sourceId;
        return this;
    }

    public Boolean isAssigned() {
        return assigned;
    }

    public MinimalSourceDetailsDTO assigned(Boolean assigned) {
        this.assigned = assigned;
        return this;
    }

    public String getSourceName() {
        return sourceName;
    }

    public MinimalSourceDetailsDTO sourceName(String sourceName) {
        this.sourceName = sourceName;
        return this;
    }

    public Map<String, String> getAttributes() {
        return attributes;
    }

    public MinimalSourceDetailsDTO attributes(Map<String, String> attributes) {
        this.attributes = attributes;
        return this;
    }

    public String getSourceTypeCatalogVersion() {
        return sourceTypeCatalogVersion;
    }

    public MinimalSourceDetailsDTO sourceTypeCatalogVersion(String sourceTypeCatalogVersion) {
        this.sourceTypeCatalogVersion = sourceTypeCatalogVersion;
        return this;
    }

    public String getSourceTypeModel() {
        return sourceTypeModel;
    }

    public MinimalSourceDetailsDTO sourceTypeModel(String sourceTypeModel) {
        this.sourceTypeModel = sourceTypeModel;
        return this;
    }

    public String getSourceTypeProducer() {
        return sourceTypeProducer;
    }

    public MinimalSourceDetailsDTO sourceTypeProducer(String sourceTypeProducer) {
        this.sourceTypeProducer = sourceTypeProducer;
        return this;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setSourceTypeId(Long sourceTypeId) {
        this.sourceTypeId = sourceTypeId;
    }

    public void setSourceTypeProducer(String sourceTypeProducer) {
        this.sourceTypeProducer = sourceTypeProducer;
    }

    public void setSourceTypeModel(String sourceTypeModel) {
        this.sourceTypeModel = sourceTypeModel;
    }

    public void setSourceTypeCatalogVersion(String sourceTypeCatalogVersion) {
        this.sourceTypeCatalogVersion = sourceTypeCatalogVersion;
    }

    public void setSourceId(UUID sourceId) {
        this.sourceId = sourceId;
    }

    public void setSourceName(String sourceName) {
        this.sourceName = sourceName;
    }

    public void setAssigned(Boolean assigned) {
        this.assigned = assigned;
    }

    public void setAttributes(Map<String, String> attributes) {
        this.attributes = attributes;
    }

    @Override
    public String toString() {
        return "MinimalSourceDetailsDTO{"
                + "id=" + id
                + ", sourceTypeId=" + sourceTypeId
                + ", sourceTypeProducer='" + sourceTypeProducer + '\''
                + ", sourceTypeModel='" + sourceTypeModel + '\''
                + ", sourceTypeCatalogVersion='" + sourceTypeCatalogVersion + '\''
                + ", expectedSourceName='" + expectedSourceName + '\''
                + ", sourceId=" + sourceId
                + ", sourceName='" + sourceName + '\''
                + ", assigned=" + assigned
                + ", attributes=" + attributes + '}';
    }
}

File: src/main/java/org/radarbase/management/service/dto/MinimalSourceDetailsDTO.kt
package org.radarbase.management.service.dto

import java.util.*

/**
 * Created by nivethika on 13-6-17.
 */
class MinimalSourceDetailsDTO {
    var id: Long? = null
    var sourceTypeId: Long? = null
    var sourceTypeProducer: String? = null
    var sourceTypeModel: String? = null
    var sourceTypeCatalogVersion: String? = null
    var expectedSourceName: String? = null
        private set
    lateinit var sourceId: UUID
    var sourceName: String? = null
    var isAssigned: Boolean? = null
    var attributes: Map<String, String> = HashMap()
    fun id(id: Long?): MinimalSourceDetailsDTO {
        this.id = id
        return this
    }

    fun sourceTypeId(sourceTypeId: Long?): MinimalSourceDetailsDTO {
        this.sourceTypeId = sourceTypeId
        return this
    }

    fun setExpectedSourceName(expectedSourceName: String?): MinimalSourceDetailsDTO {
        this.expectedSourceName = expectedSourceName
        return this
    }

    fun sourceId(sourceId: UUID): MinimalSourceDetailsDTO {
        this.sourceId = sourceId
        return this
    }

    fun assigned(assigned: Boolean?): MinimalSourceDetailsDTO {
        isAssigned = assigned
        return this
    }

    fun sourceName(sourceName: String?): MinimalSourceDetailsDTO {
        this.sourceName = sourceName
        return this
    }

    fun attributes(attributes: Map<String, String>): MinimalSourceDetailsDTO {
        this.attributes = attributes
        return this
    }

    fun sourceTypeCatalogVersion(sourceTypeCatalogVersion: String?): MinimalSourceDetailsDTO {
        this.sourceTypeCatalogVersion = sourceTypeCatalogVersion
        return this
    }

    fun sourceTypeModel(sourceTypeModel: String?): MinimalSourceDetailsDTO {
        this.sourceTypeModel = sourceTypeModel
        return this
    }

    fun sourceTypeProducer(sourceTypeProducer: String?): MinimalSourceDetailsDTO {
        this.sourceTypeProducer = sourceTypeProducer
        return this
    }

    override fun toString(): String {
        return ("MinimalSourceDetailsDTO{"
                + "id=" + id
                + ", sourceTypeId=" + sourceTypeId
                + ", sourceTypeProducer='" + sourceTypeProducer + '\''
                + ", sourceTypeModel='" + sourceTypeModel + '\''
                + ", sourceTypeCatalogVersion='" + sourceTypeCatalogVersion + '\''
                + ", expectedSourceName='" + expectedSourceName + '\''
                + ", sourceId=" + sourceId
                + ", sourceName='" + sourceName + '\''
                + ", assigned=" + isAssigned
                + ", attributes=" + attributes + '}')
    }
}
